src/main/java/org/fenixedu/cms/ui/AdminPosts.java

```diff
@@ -18,222 +18,12 @@
-
-            post.recover();
-        });
-
-        return new RedirectView("/cms/posts/" + slugSite + "/" + post.getSlug() + "/edit", true);
-    }
-
-
-    @RequestMapping(value = "{slugSite}/{slugPost}/files", method = RequestMethod.POST, produces = JSON)
-    public @ResponseBody String addFile(@PathVariable String slugSite, @PathVariable String slugPost,
-                    @RequestParam String name, @RequestParam boolean embedded, @RequestParam MultipartFile file) {
-        Site s = Site.fromSlug(slugSite);
-        Post post = s.postForSlug(slugPost);
-        ensureCanEditPost(post);
-    
-        File tmp;
-    
-        try {
-            tmp = File.createTempFile("cms-","-post");
-        } catch(IOException e) {
-            throw new RuntimeException("Error creating Post File for post " + post, e);
-        }
-    
-        try {
-            file.transferTo(tmp);
-            PostFile postFile = service.createFile(post, name, embedded, post.getCanViewGroup(), tmp);
-            return service.serializePostFile(postFile).toString();
-    
-        } catch (IOException e){
-            throw new RuntimeException("Error creating Post File for post " + post, e);
-        } finally {
-            tmp.delete();
-        }
-        
-    }
-
-    @RequestMapping(value = "{slugSite}/{slugPost}/metadata", method = RequestMethod.GET)
-    public String viewEditMetadata(Model model, @PathVariable String slugSite, @PathVariable String slugPost) {
-        Site s = Site.fromSlug(slugSite);
-        Post post = s.postForSlug(slugPost);
-        ensureCanEditPost(post);
-        PermissionEvaluation.ensureCanDoThis(s, Permission.SEE_METADATA, Permission.EDIT_METADATA);
-        model.addAttribute("site", s);
-        model.addAttribute("post", post);
-        model.addAttribute("metadata", Optional.ofNullable(post.getMetadata()).map(PostMetadata::json).map(
-            JsonElement::toString).orElseGet(()->new JsonObject().toString()));
-        return "fenixedu-cms/editMetadata";
-    }
-
-    @RequestMapping(value = "{slugSite}/{slugPost}/metadata", method = RequestMethod.POST)
-    public RedirectView editMetadata(@PathVariable String slugSite,
-                                         @PathVariable String slugPost,
-                                         @RequestParam String metadata) {
-        Site s = Site.fromSlug(slugSite);
-        Post post = s.postForSlug(slugPost);
-        FenixFramework.atomic(()-> {
-            ensureCanEditPost(post);
-            PermissionEvaluation.ensureCanDoThis(s, Permission.SEE_METADATA, Permission.EDIT_METADATA);
-            post.setMetadata(PostMetadata.internalize(metadata));
-        });
-        SiteActivity.editedPost(post, Authenticate.getUser());
-        return new RedirectView("/cms/posts/" + s.getSlug() + "/" + post.getSlug() + "/metadata", true);
-    }
-
     public static void ensureCanEditPost(Post post) {
```

Comment:
@joaofreitas-qubit this feels a bit non-domain driven.  I looked for references and it looks like it only use in this module, still being conservative, I suggest we mark both these methods as @Deprecated and implement the method directly in the post domain object.